marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx

```diff
@@ -0,0 +1,634 @@
+
+  //Documentation links for using Xpath expressions
+  const xPathDocLinks = <div className={styles.xpathDoc}><span id="doc">Documentation:</span>
+    <div><ul className={styles.docLinksUl}>
+      <li><a href="https://www.w3.org/TR/xpath/all/" target="_blank" rel="noopener noreferrer" className={styles.docLink}>XPath Expressions</a></li>
+      <li><a href="https://docs.marklogic.com/guide/app-dev/TDE#id_99178" target="_blank" rel="noopener noreferrer" className={styles.docLink}>Extraction Functions</a></li>
+      <li><a href="https://docs.marklogic.com/datahub/flows/dhf-mapping-functions.html" target="_blank" rel="noopener noreferrer" className={styles.docLink}>Mapping Functions</a></li>
+    </ul></div>
+  </div>;
+
+  const getColumnsForEntityTable:any = () => {
+    return entityColumns.map(el => props.checkedEntityColumns[el.key] ? el : "").filter(item => item);
+  };
+
+  const getValue = (object, keys) => keys.split(".").reduce((o, k) => (o || {})[k], object);
+
+  useEffect(() => {
+    initializeMapExpressions();
+    setMappingFunctions();
+    return (() => {
+      setMapExp({});
+      setSearchEntityText("");
+      setSearchedEntityColumn("");
+    });
+  }, [props.entityTypeProperties]);
+
+  const getEntityDataType = (prop) => {
+    return prop.startsWith("parent-") ? prop.slice(prop.indexOf("-")+1) : prop;
+  };
+
+  const mapExpressionStyle = (propName) => {
+    const mapStyle: CSSProperties = {
+      width: "22vw",
+      verticalAlign: "top",
+      justifyContent: "top",
+      borderColor: checkFieldInErrors(propName) ? "red" : ""
+    };
+    return mapStyle;
+  };
+
+  /*  The source context is updated when mapping is saved/loaded, this function does a level order traversal of entity
+     json and updates the sourceContext for every entity property */
+
+  const updateSourceContext = (mapExp, entityTable) => {
+    let queue:any[] = [];
+    entityTable.forEach(element => {
+      element["parentVal"] = "";
+      queue.push(element);
+    });
+
+    while (queue.length > 0) {
+      let element = queue.shift();
+      let name = element.name;
+      let parentVal = element["parentVal"];
+      if (element.hasOwnProperty("children")) {
+        if (!parentVal) {
+          tempSourceContext[name] = "";
+        } else {
+          tempSourceContext[name] = parentVal;
+        }
+        if (mapExp[name]) {
+          if (parentVal) {
+            parentVal = parentVal + "/" + mapExp[name];
+          } else {
+            parentVal = mapExp[name];
+          }
+        } else {
+          parentVal = "";
+        }
+        element.children.forEach(ele => {
+          ele.parentVal = parentVal;
+          queue.push(ele);
+        });
+      } else {
+        if (parentVal) {
+          tempSourceContext[name] = parentVal;
+        } else {
+          tempSourceContext[name] = "";
+        }
+      }
+    }
+  };
+
+  const initializeMapExpressions = () => {
```

Comment:
Looks like we are initializing Map expressions in both, the mapping-step-detail (parent) component as well as entity-map-table (child) component. It could probably be streamlined to only be initialized at either of the places and passed on to the other via state props.

Reply:
Fixed, streamlined so only the child is initializing their own map expressions.